### PR TITLE
Fix camera panning issue when camera is parented

### PR DESCRIPTION
**Pull request by 2315137135**
fix https://forum.babylonjs.com/t/camera-panning-issue-when-using-parenting/48014

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14780/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14780/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14780/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14780/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14780/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14780/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14780/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14780/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14780/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14780/merge/testResults/webgl2playwright/index.html

**Comment by RaananW** [comment]
Tested with this demo - https://playground.babylonjs.com/?snapshot=refs/pull/14780/merge#C2ZGCJ#20

File: packages/dev/core/src/Cameras/arcRotateCamera.ts [hunk 1]
```diff
@@ -963,6 +963,10 @@ export class ArcRotateCamera extends TargetCamera {
                         this._target.copyFrom(this._transformedDirection);
                     }
                 } else {
+                    if (this.parent) {
+                        const m = TmpVectors.Matrix[0];
+                        Vector3.TransformCoordinatesToRef(this._transformedDirection, this.parent.getWorldMatrix().getRotationMatrix().invertToRef(m), this._transformedDirection);
```
Comment (RaananW) [inline packages/dev/core/src/Cameras/arcRotateCamera.ts:968]: getRotationMatrix generates a new matrix, would be better to use the getRotationMatrixToRef and then invert it in place (the invert function inverts in place).

**Review by RaananW — COMMENTED** [review]
LGTM! just a small request to not generate new objects.

File: packages/dev/core/src/Cameras/arcRotateCamera.ts (hunk 1 above)
Comment (2315137135) [reply packages/dev/core/src/Cameras/arcRotateCamera.ts]: I get it.

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14780/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)
Important - these might fail sporadically. This is an optional test.

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14780/merge/testResults/webgpuplaywright/index.html

**Review by RaananW — APPROVED** [review]

**Review by sebavan — APPROVED** [review]

**Pull request merged**